File: 2024/April/4-2 205 Isomorphic Strings/IsomorphicStrings.java
import java.util.HashMap;

class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character, Character> map = new HashMap<Character, Character>();
        for(int i = 0; i < s.length(); i++) {
            if(!map.containsKey(s.charAt(i))) {
                if(!map.containsValue(t.charAt(i))) {
                    map.put(s.charAt(i), t.charAt(i));
                }
                else {
                    return false;
                }
            }
            else {
                if(map.get(s.charAt(i)) != t.charAt(i)) {
                    return false;
                }
            }
        }

        return true;
    }
}

File: 2024/April/4-2 205 Isomorphic Strings/IsomorphicStrings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        sTOt = {}

        for sChar, tChar in zip(s, t):
            if not sChar in sTOt:
                if not tChar in sTOt.values():
                    sTOt[sChar] = tChar
                else:
                    return False
            elif sTOt[sChar] != tChar:
                return False
        
        return True